MMDevice/MMDevice.h

```diff
@@ -1428,4 +1445,307 @@ namespace MM {
+       * \return  true if dataset is read-only, false if it can accept new images
+       */
+      virtual bool IsReadOnly(const char* handle) = 0;
+
+      /**
+       * \brief   Loads an existing dataset
+       * \details Loaded datasets are immutable and cannot accept new images.
+       *          Implementation may use lazy loading for efficiency.
+       *
+       * \param   path Path to the dataset
+       * \param   [out] handle Output parameter for the dataset handle
+       * \return  Status code indicating success or failure
+       */
+      virtual int Load(const char* path, char* handle) = 0;
+
+      /**
+       * \brief   Checks if the device can load a dataset at the specified path
+       * \details This will be used by the caller to quickly determine which driver to use
+       *          for a given dataset path. The implementation should ideally be very fast
+       *          and not open any files.
+       * \param   path Path to the dataset
+       * \return  true if the device can load the dataset, false otherwise
+       */
+      virtual bool CanLoad(const char* path) = 0;
+
+      /**
+       * \brief Gets the progress of the current operation
+       *
+       * \param handle Dataset handle
+       * \return Progress value (0-100), or -1 if idle or not implemented
+       * \note Assumes single operation execution at a time
+       */
+      virtual int GetProgress(const char* handle) = 0;
+
+      /**
+       * \brief   Deletes a dataset
+       *
+       * \details The dataset is permanently removed from storage.
+       * \param   handle Handle of the dataset to delete
+       * \return  Status code indicating success or failure
+       */
+      virtual int Delete(char* handle) = 0;
+
+      /**
+       * \brief   Lists datasets in a specified path
+       *
+       * \param   path Directory path to search for known datasets
+       * \param   [out] listOfDatasets Array of dataset names, each with a maximum length of maxItemLength
+       * \param   maxItems Maximum number of items to return
+       * \param   maxItemLength Maximum length of each item name
+       * \return  Status code indicating success or failure
+       */
+      virtual int List(const char* path, char** listOfDatasets, int maxItems, int maxItemLength) = 0;
+
+      /**
+       * \brief   Inserts an image at specific coordinates
+       *
+       * \param   handle Dataset handle
+       * \param   sizeInBytes Size of the image data in bytes
+       * \param   pixels Pointer to the image pixel data
+       * \param   coordinates Array of coordinates for image insertion
+       * \param   numCoordinates Number of coordinate values
+       * \param   imageMeta Image metadata string
+       * \return  Status code indicating success or failure
+       */
+      virtual int AddImage(const char* handle, int sizeInBytes, unsigned char* pixels,
+                           int coordinates[], int numCoordinates, const char* imageMeta) = 0;
```

Comment:
Sizes should be `std::size_t` or `std::uint64_t` (for memory sizes and on-disk sizes, resp.).
I think coordinates should also be 64-bit, just to be safe. `numCoordinates` can be `std::size_t`.